src/js/components/stock-movement-wizard/outbound/AddItemsPage.jsx

```diff
@@ -709,49 +735,97 @@ class AddItemsPage extends Component {
               }
-
               if (
-                _.includes(savedItemsProductCodes, item.product.productCode)
+                _.includes(savedItemsProductCodes, item.product?.productCode)
                 && parseInt(item.quantityRequested, 10) > 0
+                && item.status === SaveStatus.PENDING
               ) {
-                return { ...item, disabled: true };
+                return { ...itemToChange, disabled: true, status: SaveStatus.SAVED };
               }
-
               return item;
             });
-            this.setState({ values: { ...this.state.values, lineItems: lineItemsAfterSave } });
+            this.setState({
+              values: { ...this.state.values, lineItems: lineItemsAfterSave },
+              currentLineItems: lineItemsBackendData,
+            });
             return;
           }
 
-          this.setState({ values: { ...this.state.values, lineItems: lineItemsBackendData } });
-
           this.setState({
+            values: { ...this.state.values, lineItems: lineItemsBackendData },
             currentLineItems: lineItemsBackendData,
           });
         })
-        .catch(() => Promise.reject(new Error(this.props.translate('react.stockMovement.error.saveRequisitionItems.label', 'Could not save requisition items'))));
+        .catch(() => {
+          // When there is an error during saving we have to find products which
+          // caused the error. These items are not saved, so we don't have line ID,
+          // and we have to find these items by product ID and SaveStatus
+          const notSavedItemsIds = payload.lineItems.map(item => item['product.id']);
+          const lineItemsWithErrors = this.state.values.lineItems.map((item) => {
+            if (
+              item.product &&
+              item.status === SaveStatus.PENDING &&
+              _.includes(notSavedItemsIds, item.product.id)
+            ) {
+              return { ...item, status: SaveStatus.ERROR };
+            }
+            return item;
+          });
+          this.setState({ values: { ...this.state.values, lineItems: lineItemsWithErrors } });
+          return Promise.reject(new Error(this.props.translate('react.stockMovement.error.saveRequisitionItems.label', 'Could not save requisition items')));
+        });
     }
 
     return Promise.resolve();
   }
 
-  saveProgress = (values) => {
-    this.setState({ isSaveCompleted: false });
+  // if rowIndex is passed, it means that we are editing row
+  // not adding new one
+  saveProgress = ({ values, rowIndex, fieldValue }) => {
+    // I can't check !!rowIndex, because it can be 0,
+    // so there is possibility to return false when the
+    // rowIndex is present
+    const isEdited = rowIndex !== undefined;
+    const itemsWithStatuses = values.lineItems.map((item) => {
+      if (isEdited && rowIndex === values.lineItems.indexOf(item)) {
+        return { ...fieldValue, status: SaveStatus.PENDING };
```

Comment:
doesn't the row (an item) has also a `status` property itself, that might be useful somewhere else? If it does and you are having `status` in here, you override it.
If I'm correct, we should think about a different name for this

Reply:
As I can see there is a property `statusCode` not `status`, but maybe something like `saveStatus` or `saveProgress` will be better to avoid future complication?

Reply:
I think it could be a better option, but I'm not an expert in naming stuff

Reply:
@kchelstowski is right, having this value named `status` might lead into some complications.
We need to stay consistent so my suggestion is to name it `saveStatus` just like you named the _enum SaveStatus_

Reply:
I changed the name of this property to `saveStatus`.

Reply:
@alannadolny as I said on our "ad hoc code review meeting" words `status` and `statusCode` are "reserved".